Source/DafnyCore/Backends/GoLang/GoCodeGenerator.cs

```diff
@@ -179,9 +205,22 @@ private struct Import {
       return w;
     }
 
-    protected override void DependOnModule(string moduleName, bool isDefault, ModuleDefinition externModule,
+    protected override void DependOnModule(Program program, ModuleDefinition module, ModuleDefinition externModule,
       string libraryName) {
-      var import = CreateImport(moduleName, isDefault, externModule, libraryName);
+      var goModuleName = "";
+      if (GoModuleMode && program.Compilation.AlreadyCompiledRoots.Contains(module.Tok.Uri)) {
+        var dooManifest = DooFile.Read(module.Tok.Uri.AbsolutePath).Manifest;
```

Comment:
I definitely don't love having a code generator re-read assets from the file system, it feels like that's going to bite us in the future. But I also appreciate this is much easier than attaching doo file metadata to the AST. Can we cut an issue to revisit?

Reply:
https://github.com/dafny-lang/dafny/issues/5296